File: CS/WebDAVServer.FileSystemStorage.HttpListener/Service.cs
using System.ServiceProcess;
using System.Threading;

namespace WebDAVServer.FileSystemStorage.HttpListener
{
    partial class Service : ServiceBase
    {
        private Thread thread;
        
        public Service()
        {
            InitializeComponent();
        }
        
        #region ServiceBase Methods Override

        protected override void OnStart(string[] args)
        {
            Program.Listening = true;
            thread = new Thread(Program.ThreadProcAsync);
            thread.Start();
        }

        protected override void OnStop()
        {
            Program.Listening = false;
            thread.Abort();
            thread.Join();
        }

        #endregion // ServiceBase Methods Override

    }
}

File: VB/WebDAVServer.SqlStorage.HttpListener/Service.vb
Imports System.ServiceProcess
Imports System.Threading

Partial Class Service
    Inherits ServiceBase

    Private thread As Thread

    Public Sub New()
        InitializeComponent()
    End Sub

    Protected Overrides Sub OnStart(args As String())
        Program.Listening = True
        thread = New Thread(AddressOf Program.ThreadProcAsync)
        thread.Start()
    End Sub

    Protected Overrides Sub OnStop()
        Program.Listening = False
        thread.Abort()
        thread.Join()
    End Sub
End Class
